app/code/Magento/CatalogGraphQl/Model/Resolver/Product/SmallImageUrl.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Model\Resolver\Product;
+
+use Magento\Catalog\Helper\Image as CatalogImageHelper;
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\ResourceModel\Product\GalleryFactory as GalleryResourceFactory;
+use Magento\Framework\App\Area;
+use Magento\Framework\App\AreaList;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\Resolver\ValueFactory;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * Returns product's small image. If the small image is not set, returns a placeholder
+ */
+class SmallImageUrl implements ResolverInterface
+{
+    /**
+     * @var GalleryResourceFactory
+     */
+    private $galleryResourceFactory;
+
+    /**
+     * @var AreaList
+     */
+    private $areaList;
+
+    /**
+     * @var CatalogImageHelper
+     */
+    private $catalogImageHelper;
+
+    /**
+     * @var ValueFactory
+     */
+    private $valueFactory;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @param ValueFactory $valueFactory
+     * @param CatalogImageHelper $catalogImageHelper
+     * @param AreaList $areaList
+     * @param GalleryResourceFactory $galleryResourceFactory
+     * @param StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        ValueFactory $valueFactory,
+        CatalogImageHelper $catalogImageHelper,
+        AreaList $areaList,
+        GalleryResourceFactory $galleryResourceFactory,
+        StoreManagerInterface $storeManager
+    ) {
+        $this->valueFactory = $valueFactory;
+        $this->catalogImageHelper = $catalogImageHelper;
+        $this->areaList = $areaList;
+        $this->galleryResourceFactory = $galleryResourceFactory;
+        $this->storeManager = $storeManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ): Value {
+        if (!isset($value['model'])) {
+            $result = function () {
+                return null;
+            };
+            return $this->valueFactory->create($result);
+        }
+        /** @var Product $product */
+        $product = $value['model'];
+
+        /* If small_image is not loaded for product, need to load it separately */
+        if (!$product->getSmallImage()) {
```

Comment:
What is the use case for it to be not loaded?

Reply:
It's loaded when we specify both `small_image` and `small_image_url` in the GraphQl request. If we specify only `small_image_url` we don't have any information about the small image and cannot use `catalogImageHelper` for the image URL generation.